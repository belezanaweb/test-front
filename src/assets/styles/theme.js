export const defaultTheme = {
  borderRadius: '3px',

  fonts: {
    header: '13px',
    label: '12px',
    text: '13px',
    title: '14px',
    highlightText: '14px',
    input: '16px',
    button: '20px',
    desktop: '1rem'
  },

  colors: {
    primary: '#212122',
    secondary: '#999',
    orange: '#FF6C00',
    darkOrange: '#D45A00',
    ultraLightGray: '#E0E7EE',
    lightGray: '#CCC',
    mediumGray: '#E7E7E7',
    gray: '#EEE',
    lightBlack: '#212121',
    black: '#000',
    white: '#FFF',
    purple: '#A43287',
    red: '#F30'
  },

  shadow: '1px 1px 5px 0 rgba(0, 0, 29, 0.22)',

  breakpoints: {
    mobile: '0px',
    tablet: '576px',
    desktop: '1024px'
  }
}
