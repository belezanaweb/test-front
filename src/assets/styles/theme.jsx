export const colors = {
  background: '#eee',
  button: '#FF6C00',
  'button-hover': '#D45A00',
  'card-info-text': '#000',
  'font-color': '#212122',
  grey: '#999',
  'input-border': '#E7E7E7',
  'input-focus': '#A43287',
  'input-error': '#F30',
  orange: '#FF7800',
  placeholder: '#E0E7EE',
  'shadow-grey': '#CCC',
  white: '#fff'
}
