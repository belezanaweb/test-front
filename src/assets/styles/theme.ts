import { DefaultTheme } from 'styled-components'

const defaultTheme: DefaultTheme = {
  borderRadius: '3px',

  fonts: {
    label: '12px',
    header: '13px',
    text: '13px',
    title: '14px',
    highlightText: '14px',
    input: '16px',
    button: '20px'
  },

  colors: {
    primary: '#212122',
    secondary: '#999',
    accent: '#FF6C00',
    darkAccent: '#D45A00',
    ultraLightGray: '#E0E7EE',
    lightGray: '#CCC',
    mediumGray: '#E7E7E7',
    gray: '#EEE',
    lightBlack: '#212121',
    black: '#000',
    white: '#FFF',
    purple: '#A43287',
    red: '#F30'
  },

  shadows: {
    paper: '1px 1px 5px 0 rgba(0, 0, 29, 0.22)'
  },

  breakpoints: {
    mobile: '0px',
    tablet: '576px',
    deskop: '1024px'
  }
}

export { defaultTheme }