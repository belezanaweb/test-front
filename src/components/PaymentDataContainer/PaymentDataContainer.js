import React from 'react';
import { useSelector } from 'react-redux';

import Cardboard from '../Cardboard';
import Title from '../Title';

const PaymentDataContainer = () => {
  const { card_number, card_holder_name, card_expiration_date } = useSelector(
    (state) => state.checkout
  );

  return (
    <>
      <Title>Pagamento</Title>
      <Cardboard>
        {card_number} <br />
        {card_holder_name} <br />
        {card_expiration_date}
      </Cardboard>
    </>
  );
};

export default PaymentDataContainer;
