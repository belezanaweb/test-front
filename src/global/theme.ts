export default {
  border: {
    radius: {
      small: '3px'
    }
  },
  font: {
    family:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    weight: {
      bold: 600
    },
    size: {
      font13: '13px',
      font14: '14px',
      font20: '20px'
    },
    lineHeight: {
      lh16: '16px',
      lh17: '17px',
      lh24: '24px'
    }
  },
  color: {
    orangeButton: '#FF6C00',
    orangeText: '#FF7800',
    text: '#212122',
    textOpacity: '#cccccc',
    background: '#eeeeee',
    border: '#eeeeee',
    borderDark: '#cccccc',
    title: '#999999',
    white: '#ffffff'
  },
  container: {
    desktop: '1120px',
    tablet: '768px',
    mobile: '414px'
  }
}
