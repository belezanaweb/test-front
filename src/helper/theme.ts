export class Theme {
  primaryColor = { main: '#ff7800', contrastText: '#ffffff', dark: '#d45a00' };
  secondaryColor = { main: '#999999' };
  links = { main: '#cccccc', hover: '#bbbbbb' };
}
