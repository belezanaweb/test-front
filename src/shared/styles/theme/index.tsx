//import dark from './dark';
import light from './light';

const theme = light;
export default theme;
