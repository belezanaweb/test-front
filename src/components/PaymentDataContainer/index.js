import PaymentDataContainer from './PaymentDataContainer';

export default PaymentDataContainer;
