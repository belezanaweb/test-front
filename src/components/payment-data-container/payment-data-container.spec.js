import React from 'react';

import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';

import { create } from 'react-test-renderer';

import createMockedStore from '../../lib/test-utils/mocked-store';

import theme from '../../styles/theme';

import PaymentDataContainer from '.';

const mockedStore = createMockedStore();

const AppWrapper = ({ store = mockedStore }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <PaymentDataContainer />
    </ThemeProvider>
  </Provider>
);

describe('PaymentDataContainer', () => {
  it('should match a snapshot', () => {
    expect(create(<AppWrapper />).toJSON()).toMatchSnapshot();
  });
});
