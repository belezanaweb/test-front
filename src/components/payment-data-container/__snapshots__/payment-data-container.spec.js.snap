// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDataContainer should match a snapshot 1`] = `
Array [
  .c0 {
  color: #999;
  font-size: 0.875rem;
  font-weight: 700;
  text-transform: uppercase;
}

<h2
    className="c0"
  >
    Pagamento
  </h2>,
  .c0 {
  padding: 0.75rem;
  border-radius: 3px;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
  background-color: #fff;
}

<div
    className="c0"
  >
    ****.****.****.2222
     
    <br />
    LUAN PEREIRA
     
    <br />
    12/2021
  </div>,
]
`;
